pallets/ethereum-chain-id/src/lib.rs

```diff
@@ -18,23 +18,54 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use frame_support::{decl_module, decl_storage, traits::Get};
+use frame_support::pallet;
 
-/// Configuration trait of this pallet.
-pub trait Config: frame_system::Config {}
+pub use pallet::*;
 
-impl<T: Config> Get<u64> for Module<T> {
-	fn get() -> u64 {
-		Self::chain_id()
+#[pallet]
+pub mod pallet {
+
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	/// The Ethereum Chain Id Pallet
+	#[pallet::pallet]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	/// Configuration trait of this pallet.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	impl<T: Config> Get<u64> for Module<T> {
+		fn get() -> u64 {
+			Self::chain_id()
+		}
 	}
-}
 
-decl_storage! {
-	trait Store for Module<T: Config> as MoonbeamChainId {
-		ChainId get(fn chain_id) config(): u64 = 43;
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {}
+
+	// Is this how I give the default storage value of 43?
+	#[pallet::type_value]
+	pub fn DefaultChainId() -> u64 { 43 }
+
+	#[pallet::storage]
+	#[pallet::getter(fn chain_id)]
+	pub type ChainId<T> = StorageValue<_, u64, ValueQuery, DefaultChainId>;
+
+	#[pallet::genesis_config]
+	#[derive(Default)]
```

Comment:
note that this default means that the default value for chain_id is `0`

Reply:
I see. Should I manually `impl Default for GenesisConfig` ?

Reply:
I don't really know if the default for genesis config is sensible, but if it is then yes you can implement like this:
```rust
impl Default for GenesisConfig {
	fn default() -> Self {
		Self { chain_id: DefaultChainId::get() }
	}
}
```